src/content/1.7/modules/symfonyform/_index.md

```diff
@@ -0,0 +1,264 @@
+
+1) A form type
+2) A template to display that form type
+3) A way to handle that form.
+
+
+## Form type
+
+You can use any of Symfony's form types (https://symfony.com/doc/3.4/reference/forms/types.html) for you forms in PrestaShop. Just make sure that you pick correct version of Symfony in the documentation.
+Currently PrestaShop uses Symfony 3.4
+PrestaShop also created some custom form types, you can use them freely in your forms as well. (https://devdocs.prestashop.com/1.7/development/components/form/types-reference/).
+
+## Template
+
+In order to correctly render the form you require to get FormView of that form.
+
+First step is to build form from Form Type. In Symfony you would usually use Symfony's FormBuilder for that, but in PrestaShop it depends on what type of form you are creating.
+
+##### Settings form
+
+Settings form in PrestaShop can be created using PrestaShop\PrestaShop\Core\Form\Handler.php
+
+It still uses the same FormFactory as Symfony does, but it also sets data to the form and calls the hook specific to that form. 
+
+You cam find detailed documentation to how Settings form works here:
+https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/forms/settings-forms/
+
+And in this example module you can find an implementation:
+https://github.com/PrestaShop/example-modules/tree/master/demosymfonyform
+
+##### CRUD Form
+
+CRUD form is more complicated because it works with logic objects. 
+You can use ``PrestaShop\PrestaShop\Core\Form\IdentifiableObject\Builder\FormBuilder`` to get the form. 
+Form builder will require form type and data provider. Data provider is needed in order to fill form with data if it's an edit form. 
+Data provider must implement FormDataProviderInterface where you have functions ``getData($id)`` (retrieve data based on id) and ``getDefaultData()``.
+
+CRUD form handler:
+https://devdocs.prestashop.com/1.7/development/architecture/migration-guide/forms/crud-forms/
+
+
+After setting up form handler. You can use it to retrieve the form.
+
+```php
+$textForm = $textFormDataHandler->getForm();
+```
+
+After retrieving form you can use it to get create form view which then can be used in twig template to render said form.
+
+```php
+$this->render('@Modules/demosymfonyform/views/templates/admin/form.html.twig', [
+            'demoConfigurationTextForm' => $textForm->createView(),
+        ]);
+````
+
+In twig before PrestaShop 1.7.8 you needed to define every field for best results. 
+This allows for greater control over the form as you can define HTML exactly as you need it, but it's requires extra work if all you need is a standard form.
+
+```twig
+{% extends '@PrestaShop/Admin/layout.html.twig' %}
+
+{# PrestaShop made some modifications to the way forms are displayed to work well with PrestaShop in order to benefit from those you need to use ui kit as theme#}
+{% form_theme demoConfigurationTextForm 'PrestaShopBundle:Admin/TwigTemplateForm:prestashop_ui_kit.html.twig' %}
```

Comment:
Why form theming isn't applied by default in Symfony configuration ?
https://symfony.com/doc/4.4/form/form_themes.html#applying-themes-to-all-forms
```
# config/packages/twig.yaml
twig:
    form_themes: ['PrestaShopBundle:Admin/TwigTemplateForm:prestashop_ui_kit.html.twig']
```